okhttp/src/jvmMain/kotlin/okhttp3/internal/connection/RealConnection.kt

```diff
@@ -90,6 +92,13 @@ class RealConnection(
    * Once true this is always true. Guarded by this.
    */
   var noNewExchanges = false
+    set(value) {
```

Comment:
I expect this probably breaks our rules around not calling EventListeners inside a synchronized block?

Reply:
(but why didn’t the test fail?)

Reply:
I've added tests that catch this, will fix before landing.